Various Improvements

SkipperSkipTR opened the pull request:
## Changelog

1. **Every cutscene *(except King of the Rock last cutscene and Freedom last cutscene)* is now skipable** 
   - Closes #1

2. **Changed RC Baron Handling to make flight missions faster**  
   https://youtu.be/uEXcdt6hWwo

3. **Rewrote Fullsprint mod with toggle built-in**
   - The toggle is now more reliable. Earlier version could break when a save was loaded.
   - F8 for Fullsprint toggle and F9 for Fullcycle toggle (Fullcycle won't work without Fullsprint active)

4. **Adjusted teleport location before *King of the Rock* mission**  
   - Player is now teleported further from the mountain to ensure collision loads properly  
   - Closes #12  
   https://youtu.be/swTgpKVUIoo

5. **Updated mission progression logic for *The Terrible Claw***  
   - Adds 2 to the passed missions counter after completion  
   - Closes #25

6. **Added side script to make dinosaurs sprint toward the player in *The new government* mission**  
   - Uses `TASK_GO_STRAIGHT_TO_COORD` based on player's position  
   - Closes #14  
   https://youtu.be/9MbhPUSHyhg

7. **Timer reduced to 1 minute in *Dangerous Waters* (from 2:40)**  
   - Closes #23

8. **Added prompt for picking up the egg in *Revenge***  
   - Closes #11  
   ![Screenshot_2](https://github.com/user-attachments/assets/a7448477-8b3e-456d-980d-9da60df6e350)

9. **Updated *The Tarbos* mission logic**  
   - Now player doesn't need to go to the platform to progress the mission if the Tarbosaurus is dead.
   - Adjusted Tarbosaurus location  
   - Closes #24  
   https://youtu.be/K1EN_jcjfsA

10. **Added in-game GUI for Passed Missions Changer**  
   - Includes toggleable auto-reset option  
   ![Screenshot_1](https://github.com/user-attachments/assets/fe1f2614-09f9-4c35-9d6b-e59d14040e74)

11. **Improved death handling script**  
     - Now only sets player model if different from expected  
     - Aims to reduce softlocks  
     - Initial fix for #15

12. **Replaced font letters "k" and "y" with Latin letters**  
     - Closes #10  
     <img width="2039" height="252" alt="fontdeneme" src="https://github.com/user-attachments/assets/60602ebe-ae4f-4a7d-b062-9323f77e9143" />

13. **Created a rubberband mod that either teleports or speeds up group members**  
     - Has 3 modes: Off, Speed up 3x and Teleport
     - Can toggle between these modes by pressing F10
     - Initial implementation of #7
   https://youtu.be/JzdrXMxpsv8
   https://youtu.be/JrC3Pv7HXiE

14. **Edited `gta_sa.exe` file**  
     - Changed User Files folder name to: `CretaceousRunner SRV Files`  
     - Changed `.set` file name to: `cr_srv.set`
     - Adds to #3 

15. **Fixed some bugs**
     - The bug in *Welcome to the ghetto* where if you kill the Hadrosaurus with your last rock, the mission would fail.
     - The bug in *Grey hills* where if you fail the mission, *In the dream* would start.

16. **Enemy markers now get cleared after killing them in *Street Fighting* and *Dayarhell***
![1](https://github.com/user-attachments/assets/c68b93b7-749d-4ec4-a4f7-2dfaab74af95)
![2](https://github.com/user-attachments/assets/397f9b62-a083-4987-9100-b4ec9827d764)

SkipperSkipTR committed:
Added an in-game GUI for Passed Missions Changer.
It can be opened with F7. Also added configuration for changing the passed missions value to zero every time game is restarted.

SkipperSkipTR committed:
Fixed the Passed Missions Changer GUI for auto-reset.

SkipperSkipTR committed:
Changed handling to make RC Baron faster.

SkipperSkipTR committed:
Fix for King of the Rock mountain collision.
Also added cutscene skips to the mission.

SkipperSkipTR committed:
Passed_Missions=32 non-mission is skipped now.

SkipperSkipTR committed:
Dayargolia cutscene is made skipable.

SkipperSkipTR committed:
Fix the buttons on the Passed Missions Changer GUI not being the proper size.

SkipperSkipTR committed:
Made dinosaurs in The New Government sprint
Also they now follow the player correctly and teleport if they get stuck.

SkipperSkipTR committed:
Timer in Dangerous Waters reduced to 1 minute
Added a cutscene skip at the beginning of the mission.

SkipperSkipTR committed:
Added cutscene skip to the ending cutscene of Dangerous Waters

SkipperSkipTR committed:
Added a prompt for picking up the egg in Revenge

SkipperSkipTR committed:
Updated the cleo text with the prompt

SkipperSkipTR committed:
Changed tarbos location in The Tarbos
Also removed the requirement to go the marker before mission actually ends. If tarbos dies the mission is instantly complete.

SkipperSkipTR committed:
Framerate Vigilante now limits to 30 instead of 25
As requested by Josh

SkipperSkipTR committed:
Added cutscene skip to Flying School

SkipperSkipTR committed:
Changed unnecessary global variables to local in external scripts
I really thought $ was used for all variables. I was wrong...

SkipperSkipTR committed:
Fixed Full Sprint Toggle not working properly with game restarts.

SkipperSkipTR committed:
Added cutscene skip and made hodry sprint in "In the Dream"
Also changed death script to only change the player model, if it is different from expected model

SkipperSkipTR committed:
Edited cyrillic letters in font.

DromeoStalker commented:
Did you ckeck how RC baron in Future depends on you works now?

SkipperSkipTR commented:
> Did you ckeck how RC baron in Future depends on you works now?

I did now. I encountered 2 problems;
1. Not being able to pull up when going too fast. (I think this might be caused by being too high up.)
2. Dying because of the task_leave_any_car command. Some plane angles were just killing the player. 

I think we can just replace the task_leave_any_car with warp_char_from_car_to_coord and add 5 to player's x or y coordinates to fix the mission:

https://youtu.be/YTiIV4IRXCQ

SkipperSkipTR committed:
Changed "Future Depends on You" to work with new RC Baron handling

SkipperSkipTR committed:
Small correction for last commit

DromeoStalker commented:
> > Did you ckeck how RC baron in Future depends on you works now?
> 
> I did now. I encountered 2 problems;
> 
> 1. Not being able to pull up when going too fast. (I think this might be caused by being too high up.)
> 2. Dying because of the task_leave_any_car command. Some plane angles were just killing the player.
> 
> I think we can just replace the task_leave_any_car with warp_char_from_car_to_coord and add 5 to player's x or y coordinates to fix the mission:
> 
> https://youtu.be/YTiIV4IRXCQ

Looks almost OK, maybe a bit slower speed make this all work better.

DromeoStalker commented:
Killing because of too long exit animation and despawn of plane, little longer time will fix this. Kinda crash cuz didn't land carefully.

SkipperSkipTR commented:
> Killing because of too long exit animation and despawn of plane, little longer time will fix this. Kinda crash cuz didn't land carefully.

Oh yeah, I didn't see the car being deleted. Adding this loop to check for being on foot solved the dying issue.
![Screenshot_1](https://github.com/user-attachments/assets/7be91d3f-52e4-4f0c-8110-fdb5af840633)

I think speed of the baron is fine though.

SkipperSkipTR committed:
Added a better fix for the "Future depends on you"

DromeoStalker commented:
Yea, without deleting, the sound of wind (engine) is keep going, so the plane was changed to static model. Also try after first checkpoint fly to the water, it will be fun. I think it's possible to make there mission fail by checking height of the player, about meter of two lower than airport surface. Damn, after 12 years, I still remember how this crap works.

SkipperSkipTR committed:
Added cutscene skips to intro missions
Added Childhood fix to the group member fixes

SkipperSkipTR commented:
Added cutscene skips to:
- Introduction
- Home and Native Land
- Cantera
- Childhood
- Catch me who can
- Surprise Dangerous
- Nafta Rewriting

Also made a new group member fix script that gets the players coordinates and makes all the group members sprint to it (See the screenshot on the main changelog). It can be applied to multiple missions. Need to find other missions that can use this.

DromeoStalker commented:
Wtf is happened to mount chiliad collision? Never happened shit like that.

SkipperSkipTR committed:
Changed the music starting position in Mating Dance

Added cutscene skips to Chepei, Other Lands, Coming Home, Welcome to Ghetto, Death from the Sky, Frontier Guard and Back to Alfred missions

Added char_clear_tasks to cutscene skips in Dayarhell and The Crocodiles missions

SkipperSkipTR committed:
Changed save file location to CretaceousRunner SRV Files
Changed .set name to cr_srv.set

SkipperSkipTR committed:
Added cutscene skips to all Cantera chapter missions
New cutscene skips: Feathers is a sign of skill, New opportunities, Save the Life, Life's beach, 6-year War, Street Fighting, Quieter than water, The Great Cause, The Big Brother
Added Street Fighting and Fishing missions to the Group Member Fixes side script. Which makes every group member sprint to player's position.

SkipperSkipTR committed:
Added cutscene skips to more missions
Grey Hills, Candyland, Snowboarding, The Alchemist, Hello, North!, Cold Wind, Fishing, The Great Escape, Nowhere to run, Amarhaan, Egg Stiffs, The terrible claw, Weaponwood, Cretaceous Running, I wanna bite, Density  Post, Future depends on You, Progenesis, The Promenade, Like a First Time, Evil Return, Exploration, More than Business, Fireman, Grilled Chicken, The New Government and Freedom (except last cutscene) now have skipable cutscenes

SkipperSkipTR commented:
With the last commit, all missions should have cutscene skips now (except the very last cutscene)

I've tested them one by one while adding them but needs a full game test as well.

SkipperSkipTR committed:
Fixed failing "Grey Hills" putting player back to "In the Dream"

SkipperSkipTR committed:
Added and modified cutscene skips in July 1st

SkipperSkipTR committed:
Added cutscene skip to The Prankster

SkipperSkipTR committed:
Fixed a bug in "Welcome to the ghetto"
Where if you kill Hadrosaurus with your last rock, you fail the mission

SkipperSkipTR committed:
Moved player to the correct position after skipping the cutscene in Frontier Guard
Also added a cutscene skip

SkipperSkipTR commented:
Did a full run, tried to fix things I noticed. I think its ready to merge.

SkipperSkipTR committed:
Added a cutscene skip to Nowhere to Run
Fixed savefile name not changing if you skip the ending cutscene of dangerous waters

SkipperSkipTR committed:
Enemy markers in Street Fighting now gets removed when they die

SkipperSkipTR committed:
Added fade in and random starting time to music script

SkipperSkipTR committed:
Enemy markers in Dayarhell now get removed when they die

SkipperSkipTR committed:
Rewrote Fullsprint mod and added a toggle.
Old toggle script was buggy, so I rewrote the mod and added the toggle natively.

SkipperSkipTR committed:
Changed autosplitter to correctly work with removed 32nd mission

SkipperSkipTR committed:
Changed split file formatting

SkipperSkipTR committed:
Added a way to toggle FullCycle, disabled by default

SkipperSkipTR committed:
Added some missing skip_cutscene_end commands to the main script

SkipperSkipTR committed:
Started fixing issues when not skipping the cutscenes with the added cutscene skips
Also fixed the group member fixes script. It wasn't getting the player group correctly before.

SkipperSkipTR committed:
Added a group member rubberband mod that can be toggled with F10
Has 3 modes: Off, Speed up 3x, Teleport
Also removed the sprint command from group member fixes mod.

SkipperSkipTR committed:
Fixed some issues that could occur when not skipping the cutscenes

SkipperSkipTR committed:
Music now fades in correctly

SkipperSkipTR committed:
Added a heal to Group Member Rubberband mod when using Speed 3x mode.
This is to make sure player's followers don't die from fall damage when speeding up.

SkipperSkipTR committed:
Fixed weapon icon showing a white box.

Fixed a softlock in Freedom where skipping the cutscene after player control is set to false, game doesn't return the control back.

SkipperSkipTR committed:
Made all music loop since some of them are shorter than their respective mission's length.
